Rewrite SVG slope page (#34767)
mdn/content: files/en-us/web/svg/attribute/intercept/index.md

title: intercept
slug: Web/SVG/Attribute/intercept
page-type: svg-attribute
browser-compat: svg.elements.feFuncR
spec-urls: https://drafts.fxtf.org/filter-effects/#element-attrdef-fecomponenttransfer-intercept
---

{{SVGRef}}

> The **`intercept`** attribute defines the intercept of the linear function of color component transfers when the {{SVGAttr("type")}} attribute is set to `linear`. When the `type` of a transfer function is `linear`, the `intercept` defines the intercept of the linear function. The `linear` function is defined as `slope * color + intercept`, where `color` is the color value.

Author: ```suggestion
The **`intercept`** attribute defines the intercept of the linear function of color component transfers when the {{SVGAttr("type")}} attribute is set to `linear`. When the `type` of a transfer function is `linear`, the `intercept` defines the intercept of the linear function. The `linear` function is defined as `slope * color + intercept`, where `color` is the color value, the `intercept` provides a base value for the result, and the `slope` is a scaling factor.
```

> The **`intercept`** attribute defines the intercept of the linear function of color component transfers when the {{SVGAttr("type")}} attribute is set to `linear`. The `linear` function is defined as `slope * color + intercept`, where `color` is the color value, the `intercept` provides a base value for the result, and the [`slope`](/en-US/docs/Web/SVG/Attribute/slope) is a scaling factor.

Reviewer: FWIW I prefer that the first sentence is like a stand-alone abstract. It makes reader's lives easier because they can read just that first sentence to understand if they are interested. NOt blocking acceptance of this on that though.

```suggestion
The **`intercept`** attribute defines the intercept of the linear function of color component transfers when the {{SVGAttr("type")}} attribute is set to `linear`.

The `linear` function is defined as `slope * color + intercept`, where `color` is the color value, the `intercept` provides a base value for the result, and the [`slope`](/en-US/docs/Web/SVG/Attribute/slope) is a scaling factor.
```

> The **`intercept`** attribute defines the intercept of the linear function of color component transfers when the {{SVGAttr("type")}} attribute is set to `linear`. When the `type` of a transfer function is `linear`, the `intercept` defines the intercept of the linear function. The `linear` function is defined as `slope * color + intercept`, where `color` is the color value, the `intercept` provides a base value for the result, and the `slope` is a scaling factor.

Reviewer 2: ```suggestion
The **`intercept`** attribute defines the intercept of the linear function of color component transfers when the {{SVGAttr("type")}} attribute is set to `linear`. The `linear` function is defined as `slope * color + intercept`, where `color` is the color value, the `intercept` provides a base value for the result, and the [`slope`](/en-US/docs/Web/SVG/Attribute/slope) is a scaling factor.
```

> The **`intercept`** attribute defines the intercept of the linear function of color component transfers when the {{SVGAttr("type")}} attribute is set to `linear`. The `intercept` attribute is supported by children of the {{SVGElement("feComponentTransfer")}} filter primitive, including the `<feFuncR>`, `<feFuncB>`, `<feFuncG>`, and `<feFuncA>` transfer functions. When the `type` of a transfer function is `linear`, the `intercept` defines the intercept of the linear function.

Reviewer: What something is supported by should be separated from what it does.
```suggestion
The **`intercept`** attribute defines the intercept of the linear function of color component transfers when the {{SVGAttr("type")}} attribute is set to `linear`.

The `intercept` attribute is supported by children of the {{SVGElement("feComponentTransfer")}} filter primitive, including the `<feFuncR>`, `<feFuncB>`, `<feFuncG>`, and `<feFuncA>` transfer functions. When the `type` of a transfer function is `linear`, the `intercept` defines the intercept of the linear function.
```

title: intercept
slug: Web/SVG/Attribute/intercept
page-type: svg-attribute
browser-compat: svg.elements.feFuncR
spec-urls: https://drafts.fxtf.org/filter-effects/#element-attrdef-fecomponenttransfer-intercept
---

{{SVGRef}}

> The **`intercept`** attribute defines the intercept of the linear function of color component transfers when the {{SVGAttr("type")}} attribute is set to `linear`.

Reviewer: I know this predates you, but this says that "the intercept defines the intercept". What's an intercept?

Author: i have no clue, but i added the equation.

> The `intercept` attribute is supported by children of the {{SVGElement("feComponentTransfer")}} filter primitive, including the `<feFuncR>`, `<feFuncB>`, `<feFuncG>`, and `<feFuncA>` transfer functions. When the `type` of a transfer function is `linear`, the `intercept` defines the intercept of the linear function.

Author: ```suggestion
The `intercept` attribute is supported by children of the {{SVGElement("feComponentTransfer")}} filter primitive, including the `<feFuncR>`, `<feFuncB>`, `<feFuncG>`, and `<feFuncA>` transfer functions. When the `type` of a transfer function is `linear`, the `intercept` defines the intercept of the linear function. The `linear` function is defined as `slope * color + intercept`, where `color` is the color value.
```

You can use this attribute with the following SVG component transfer function elements when `type="linear"` is set:

- {{SVGElement("feFuncA")}}
- {{SVGElement("feFuncB")}}
- {{SVGElement("feFuncG")}}

> - {{SVGElement("feFuncR")}}

Reviewer 2: Doesn't this duplicate the paragraph above?
